src/transformers/models/vit/image_processing_vit.py

```diff
@@ -242,7 +242,12 @@ def preprocess(
         )
 
         if do_convert_rgb:
-            images = [convert_to_rgb(image) for image in images]
+            if not input_data_format == ChannelDimension.NONE:
+                images = [convert_to_rgb(image) for image in images]
+            # handle HW PIL images to HWC by conversion to RGB - 3 channel allows for normalization with imagenet mean
+            else:
+                images = [convert_to_rgb(np.stack([image] * 3, axis=-1)) for image in images]
+                input_data_format = ChannelDimension.LAST
 
```

Comment:
We should avoind doing this, insted we should extend `convert_to_rgb` function to handle NONE format